### ISSUE TITLE
12.0 PFConnector-Remote Possible Documentation issue or instructions issue..

### ISSUE THREAD

**Issue by PureOxygen-Dev** [inverse-inc/packetfence#7401]
**Describe the bug**
The instructions for updating pfConnector-remote results in an error.   

**To Reproduce**
1. Go to section in documentation  https://www.packetfence.org/doc/PacketFence_Installation_Guide.html#_upgrade_for_version_prior_to_12_1_0)
2. follow instructions

Here is the transcript of the error
root@NLNIJJOTAUT001:/usr/local# echo 'deb http://inverse.ca/downloads/PacketFence/debian/{release_minor} bullseye bullseye' > \
/etc/apt/sources.list.d/packetfence-pfconnector-remote.list
root@NLNIJJOTAUT001:/usr/local# apt update
Get:1 http://security.debian.org/debian-security bullseye-security InRelease [48.4 kB]
Hit:2 http://deb.debian.org/debian bullseye InRelease                                                                   
Get:3 http://deb.debian.org/debian bullseye-updates InRelease [44.1 kB]
Get:4 http://security.debian.org/debian-security bullseye-security/main Sources [172 kB]
Get:5 http://security.debian.org/debian-security bullseye-security/main amd64 Packages [208 kB]
Get:6 http://security.debian.org/debian-security bullseye-security/main Translation-en [135 kB]
Ign:7 http://inverse.ca/downloads/PacketFence/debian/{release_minor} bullseye InRelease
Err:8 http://inverse.ca/downloads/PacketFence/debian/{release_minor} bullseye Release
  404  Not Found [IP: 192.95.20.194 80]
Get:9 http://deb.debian.org/debian bullseye-updates/main Sources.diff/Index [15.1 kB]
Get:10 http://deb.debian.org/debian bullseye-updates/main amd64 Packages.diff/Index [15.1 kB]
Get:11 http://deb.debian.org/debian bullseye-updates/main Translation-en.diff/Index [6,117 B]
Get:12 http://deb.debian.org/debian bullseye-updates/main Sources T-2022-10-31-2015.41-F-2022-10-15-2035.13.pdiff [2,196 B]
Get:12 http://deb.debian.org/debian bullseye-updates/main Sources T-2022-10-31-2015.41-F-2022-10-15-2035.13.pdiff [2,196 B]
Get:13 http://deb.debian.org/debian bullseye-updates/main amd64 Packages T-2022-10-31-2015.41-F-2022-10-15-2035.13.pdiff [10.1 kB]
Get:13 http://deb.debian.org/debian bullseye-updates/main amd64 Packages T-2022-10-31-2015.41-F-2022-10-15-2035.13.pdiff [10.1 kB]
Get:14 http://deb.debian.org/debian bullseye-updates/main Translation-en T-2022-10-21-2017.32-F-2022-10-21-2017.32.pdiff [2,494 B]
Get:14 http://deb.debian.org/debian bullseye-updates/main Translation-en T-2022-10-21-2017.32-F-2022-10-21-2017.32.pdiff [2,494 B]
Reading package lists... Done   
E: The repository 'http://inverse.ca/downloads/PacketFence/debian/{release_minor} bullseye Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
root@NLNIJJOTAUT001:/usr/local# apt install -y -o Dpkg::Options::="--force-confnew" packetfence-pfconnector-remote
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package packetfence-pfconnector-remote

**Expected behavior**
Update does not through a warning or Error.

**Desktop (please complete the following information):**
 - OS: Debian Bullseye

**Comment by julsemaan** [comment]
It looks like the `{release_minor}` is not expanded correctly. As a workaround, replace it with `12.1` and it will work

I'll check with @nqb why that isn't being replaced

**Comment by PureOxygen-Dev** [comment]
Hi Julien

Looks like we are both multitasking with issues.

This also did not work it appears

root@NLNIJJOTAUT001:/usr/local# echo 'deb http://inverse.ca/downloads/PacketFence/debian/12.1 bullseye bullseye' > /etc/apt/sources.list.d/packetfence-pfconnector-remote.list
root@NLNIJJOTAUT001:/usr/local# apt update
Hit:1 http://security.debian.org/debian-security bullseye-security InRelease
Hit:2 http://deb.debian.org/debian bullseye InRelease
Hit:3 http://deb.debian.org/debian bullseye-updates InRelease
Get:4 http://inverse.ca/downloads/PacketFence/debian/12.1 bullseye InRelease [4,527 B]
Err:4 http://inverse.ca/downloads/PacketFence/debian/12.1 bullseye InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY CB2D3A2AA0030E2C
Reading package lists... Done
W: GPG error: http://inverse.ca/downloads/PacketFence/debian/12.1 bullseye InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY CB2D3A2AA0030E2C
E: The repository 'http://inverse.ca/downloads/PacketFence/debian/12.1 bullseye InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
root@NLNIJJOTAUT001:/usr/local# apt install -y -o Dpkg::Options::="--force-confnew" packetfence-pfconnector-remote
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package packetfence-pfconnector-remote
root@NLNIJJOTAUT001:/usr/local#

**Comment by julsemaan** [comment]
Looks like the upgrade instructions are also missing the pubkey setup:

Do:
```
apt update && apt install gnupg sudo
echo 'deb http://inverse.ca/downloads/PacketFence/debian/12.1 bullseye bullseye' > \
/etc/apt/sources.list.d/packetfence-pfconnector-remote.list
wget -q -O - https://inverse.ca/downloads/GPG_PUBLIC_KEY | sudo apt-key add -
apt update
apt install -y -o Dpkg::Options::="--force-confnew" packetfence-pfconnector-remote
```

That should do the trick

**Comment by PureOxygen-Dev** [comment]
Hi Julien

I have to split this line as it needed a Y to be pressed and did not appear when I tried it together

apt update && apt install gnupg sudo

But this is solved.  Maybe a couple more text changes in the doc needed :-)

Thanks for the help.

**Comment by nqb** [comment]
I will update documentation to fix issues.

`{release_minor}` should have been replaced because it was correctly set when [building doc at release](https://gitlab.com/inverse-inc/packetfence/-/jobs/3361999761#L231). I will investigate that part.

**Comment by nqb** [comment]
> Looks like the upgrade instructions are also missing the pubkey setup:

@julsemaan, not agree on that part. If you are doing an upgrade, you already installed public key when you performed initial installation (using [Installation instructions](https://www.packetfence.org/doc/PacketFence_Installation_Guide.html#_installation_4)).

@RHDHV-simon-sutcliffe can confirm but I think he didn't follow "Installation" instructions.

I only pushed a fix to have correct substitution for `release_minor`. I'm backporting that fix to maintenance/12.1 and I will rebuild doc accordingly.

**Comment by PureOxygen-Dev** [comment]
@nqb We installed the connector a couple of days after the release of 12.0.  I am not 100% sure as I have had a few G&T's since then, but this was not installed at that time how it is now documented in the 12.1 release.

That is where the issue came from about the public key I suspect as I seem to remember we installed it directly from Inverse not using the atp solution.  But as I said I am not 100% sure.
